aten/src/ATen/Context.cpp

```diff
@@ -188,6 +188,19 @@ bool Context::hasMKLDNN() {
 #endif
 }
 
+bool Context::hasMPS() {
+#if defined(__APPLE__) and defined(TARGET_ON_MAC)
+  if (@available(macOS 12.3, *)) {
+    return c10::impl::hasDeviceGuardImpl(at::DeviceType::MPS);
+  } else {
+    TORCH_WARN("MPS device is available macOS 12.3+");
```

Comment:
I'm not sure this is the best place to warn for this? This function might be called at startup and so would unconditionally warn?

Reply:
I see, can you suggest a better place where we can put it? For now i will remove it and in the later commit I can add the warning.

Reply:
👍